### PR TITLE
COMMERCE-11516 Adding onAfterRemove listener for AccountEntry to unlink CommerceCatalogs and CommerceChannels after account deletion

### PR DESCRIPTION
**Pull request by jaydawu**
https://issues.liferay.com/browse/COMMERCE-11516
Resent from https://github.com/liferay-commerce/liferay-portal/pull/3881 for an internal review.

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 1]
```diff
@@ -75,4 +75,37 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-11516. If a catalog with a supplier account is created, but the supplier account is deleted, the previously created supplier account will not be linked to the catalog via API"
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:78]: When a supplier account is deleted after being linked to a catalog, the account will not be shown as linked to the catalog.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 2]
```diff
@@ -75,4 +75,37 @@ definition {
...
+	@priority = 3
+	test CannotViewADeletedSupplierAccountLinkToACatalogViaAPI {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:80]: CannotViewDeletedSupplierAccountLinkedToCatalogViaAPI

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 3]
```diff
@@ -75,4 +75,37 @@ definition {
...
+		property portal.acceptance = "false";
+
+		task ("Given a 'Supplier' type of account is post and passed via API") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:83]: "Given a 'Supplier' type of account is created"

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 4]
```diff
@@ -75,4 +75,37 @@ definition {
...
+			CommerceJSONAccountsAPI._addCommerceAccount(
+				accountName = "Supplier Account",
+				accountType = "Supplier",
+				externalReferenceCode = "Supplier Account ERC");
+		}
+
+		task ("And a catalog with the supplier account is post and passed via API") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:90]: "And a catalog is created and the supplier account is linked to it"

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 5]
```diff
@@ -75,4 +75,37 @@ definition {
...
+			CommerceJSONCatalogsAPI._addCatalog(
+				accountEntryId = ${accountEntryId},
+				catalogName = "Test Catalog");
+		}
+
+		task ("When the supplier account is deleted") {
+			CommerceJSONAccountsAPI._deleteAllCommerceAccounts();
+		}
+
+		task ("Then the supplier account is not linked to the catalog created before") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:102]: "Then the supplier account will not be shown as linked to the catalog"

**Review by baharturk — CHANGES_REQUESTED** [review]
Hi @jaydawu, I left some comments, please take a look. This bug is fixing also the Channels, please add another test also for channels to cover that scenario. Thanks!

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:stop

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by victoroscuro** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
Hi @victoroscuro please force-forward this since there are no related errors to this pull, thanks!

**Comment by victoroscuro** [comment]
ci:forward:force

**Pull request closed without merging**